src/cla/tcpcl/net.rs

```diff
@@ -0,0 +1,182 @@
+                writer.write_u8(sess_term_data.flags.bits()).await?;
+                writer.write_u8(sess_term_data.reason as u8).await?;
+            }
+            TcpClPacket::XferSeg(xfer_seg_data) => {
+                writer.write_u8(MessageType::XferSegment as u8).await?;
+                writer.write_u8(xfer_seg_data.flags.bits()).await?;
+                writer.write_u64(xfer_seg_data.tid).await?;
+                if xfer_seg_data.flags.contains(XferSegmentFlags::START) {
+                    writer.write_u32(0).await?;
+                }
+                writer.write_u64(xfer_seg_data.len).await?;
+                writer.write_all(xfer_seg_data.buf.as_ref()).await?;
+            }
+            TcpClPacket::XferAck(xfer_ack_data) => {
+                writer.write_u8(MessageType::XferAck as u8).await?;
+                writer.write_u8(xfer_ack_data.flags.bits()).await?;
+                writer.write_u64(xfer_ack_data.tid).await?;
+                writer.write_u64(xfer_ack_data.len).await?;
+            }
+            TcpClPacket::XferRefuse(xfer_refuse_data) => {
+                writer.write_u8(MessageType::XferRefuse as u8).await?;
+                writer.write_u8(xfer_refuse_data.reason as u8).await?;
+                writer.write_u64(xfer_refuse_data.tid).await?;
+            }
+            TcpClPacket::KeepAlive => {
+                writer.write_u8(MessageType::Keepalive as u8).await?;
+            }
+            TcpClPacket::MsgReject(msg_reject_data) => {
+                writer.write_u8(MessageType::MsgReject as u8).await?;
+                writer.write_u8(msg_reject_data.reason as u8).await?;
+                writer.write_u8(msg_reject_data.header).await?;
+            }
+        }
+        Ok(())
+    }
+
+    pub async fn parse(reader: &mut (impl AsyncRead + Unpin)) -> anyhow::Result<Self, TcpClError> {
+        debug!("Parsing incoming packet");
+        let mtype = reader.read_u8().await?;
+        if let Some(mtype) = MessageType::from_u8(mtype) {
+            match mtype {
+                MessageType::XferSegment => {
+                    let flags = XferSegmentFlags::from_bits_truncate(reader.read_u8().await?);
+                    let tid: u64 = reader.read_u64().await?;
+                    if flags.contains(XferSegmentFlags::START) {
+                        let ext_len: u32 = reader.read_u32().await?;
+                        if ext_len != 0 {
+                            warn!("transfer extension are unsupported, ignoring them!");
+                        }
+                        for _ in 0..ext_len {
+                            reader.read_u8().await?;
+                        }
+                    }
+                    let len = reader.read_u64().await?;
+                    debug!("Reading xfer segment with len {}", len);
+                    let mut data = Vec::with_capacity(len as usize);
+                    data.resize_with(len as usize, Default::default);
```

Comment:
why the resize directly after creation?

Reply:
because the vector has `len() = 0` directly after creation, regardless of its maximum capacity